src/rpc/blockchain.cpp

```diff
@@ -1972,21 +1979,29 @@ static UniValue getblockstats(const JSONRPCRequest& request)
             }
             CAmount tx_total_in = 0;
             for (const CTxIn& in : tx->vin) {
-                CTransactionRef tx_in;
-                uint256 hashBlock;
-                if (!GetTransaction(in.prevout.hash, tx_in, Params().GetConsensus(), hashBlock, false)) {
-                    throw JSONRPCError(RPC_INTERNAL_ERROR, std::string("Unexpected internal error (tx index seems corrupt)"));
+                if (!in.IsZerocoinSpend()) {
+                    CTransactionRef tx_in;
+                    uint256 hashBlock;
+                    if (!GetTransaction(in.prevout.hash, tx_in, Params().GetConsensus(), hashBlock, false)) {
```

Comment:
@CaveSpectre11 any particular reason to use `false` for `fAllowSlow` transaction lookup?

Reply:
The `GetTransaction()` call is unchanged from the previous code.  It's just been conditionalized to not be called if the transaction was a zerocoin spend.

I didn't make a note of the failure path through GetTransaction, but I seem to recall it was coming back on the g_txindex condition:

```
        if (g_txindex) {
            return g_txindex->FindTx(hash, hashBlock, txOut);
        }
```

Which, of course, raises the question of when would g_txindex not exist?   In other words, I don't think the fAllowSlow condition is ever reached, so true, or false, won't actually make a difference.

Reply:
`g_txindex` is on by default you are correct. You are still able to turn off the txindex using a startup arg though. Not important for this particular code since it is not consensus based.